### ISSUE TITLE
Crash on OSX Ventura 13.4

### ISSUE THREAD

**Issue by kingdonb** [rabbit-shocker/rabbit#156]
Hello, a while ago I opened #142 and since then I've been able to use Rabbit on MacOS with success 🎉 

However, once in a while when the window shape changes, I get this crash:

```
The crash report is too long, so I posted it here in a gist:
```

https://gist.github.com/kingdonb/a6c414870bc3b4060206e887ee2dc8ac

I think this is just a little bit inconvenient, I miss the "open a hole in the slides" feature which doesn't seem to work on other platforms than X11, but the risk that slides will crash when going to or from full-screen really has only one major impact other than minor disruption to the flow of the talk, the problem is the timer getting reset. So I'm not sure what can be done about this, other than to try and fix the crash.

But other than this one minor bug, Rabbit is now completely usable for me on MacOS 😄 I'm enjoying the new Mermaid diagrams!

**Comment by kou** [comment]
Thanks for sharing the backtrace.
It seems that it's a rcairo and Ruby's GC related problem.

What were you doing when Rabbit was crashed? (Does "the window shape changes" mean that you changed window size?)

Are you using GTK with X11 backend on macOS, right?

**Comment by kingdonb** [comment]
No, I'm using Ruby on MacOS and the GTK is compiled with Homebrew. Yes, that's what I mean, the window changed dimensions, not shape. (Shape changes aren't possible without X11 from what I can tell...)

I used to use Rabbit in a VM, on a Linux environment with acceleration. That stopped working at some point due to an upgrade, I think, but at that time I decided to try on native MacOS again. And much to my surprise and delight, it doesn't have a lot of issues. (I used this on DevOps Toolkit yesterday: https://youtu.be/4tRYJ0mZf5A?t=1786)

**Comment by kou** [comment]
> I'm using Ruby on MacOS and the GTK is compiled with Homebrew.

OK. Rabbit will use GTK4 instead of GTK3 soon. (I hope...) Then the problem may be fixed.

> not for "make a hole in the slides" at least 

Recently, I don't use the "Rabbit hole" feature because it doesn't work with remote presentation when I share only the Rabbit window. (It work when I share my desktop but I don't want to do it for security concern.)

Instead, I'm using embedded terminal feature. It works with remote presentation.

https://youtu.be/C_-r8IQLAVY?t=551

> I used this on DevOps Toolkit yesterday: https://youtu.be/4tRYJ0mZf5A?t=1786

Wow! You're using a custom theme!

**Comment by kingdonb** [comment]
I included the lightning style, to be able to flip back and forth between lightning style in the same presentation.

This is really the best slide tool to use with markdown, I only wish that it also had all the slow-reveal features of Rabbitdown!

No other Flux maintainers are Rubyists for now 😸

**Comment by kingdonb** [comment]
> https://gitlab.com/ktou/rabbit-slide-kou-rubykaigi-2022

I went digging through your slides looking for how you embedded that terminal there in Rabbit, but I didn't see it!

That is a pretty cool trick 😁 after I made my Rabbit theme to perfectly match the corporate branding guidelines, a few days later Marketing came out and released a whole new set of branding guidelines, all slightly different than before! I resolved to not fix it then, because it took quite a while to figure all that out and I was exhausted, but maybe I should go back and update it to match 😅

**Comment by kou** [comment]
You need to install vte3 gem explicitly for the embedded terminal feature. :-)

**Comment by kou** [comment]
> I only wish that it also had all the slow-reveal features of Rabbitdown!

Does it mean that a slide transition feature is missing in Rabbit?

**Comment by kingdonb** [comment]
I don't think it is a missing feature, just feature parity, as I think it only works if you use rabbitdown. Is there a markdown syntax for "show these bullets in slow reveal" like `(('wait'))` in rd? I only found it in `rabbit.rd` sample, but it wasn't repeated in the `.md` file.

**Comment by kingdonb** [comment]
I did need a hint to install the `vte3` gem, thanks for that but it doesn't appear to have done anything.

Does there need to be support from the theme as well, or do you need to be on a "terminal" page? I thought there might be some keyboard mismatch because Apple keyboard usually uses command and not control key, but the "option" key appears to function as normal Alt (and it works with Alt+t to reset the timer) - i'm pressing Ctrl+Alt+Shift+T

I think it's not appropriate for me to ask all these questions on an unrelated issue, but I don't want to spam your repo with minor issues and I appreciate your time. Thank you for your assistance with understanding 🙏

**Comment by kou** [comment]
You can use `{::wait/}` in Markdown. :-)
See also: https://raw.githubusercontent.com/rabbit-shocker/rabbit/master/sample/rabbit-en.md

Ah, could you apply the following patch to the vte3 gem?

```diff
diff --git a/vte3/lib/vte3.rb b/vte3/lib/vte3.rb
index de6c0ff23..44fe90d44 100644
--- a/vte3/lib/vte3.rb
+++ b/vte3/lib/vte3.rb
@@ -27,5 +27,6 @@ module Vte
 
   Gtk.init if Gtk.respond_to?(:init)
   loader = Loader.new(self)
+  loader.version = "2.91"
   loader.load("Vte")
 end
```

We need to specify the version explicitly to use GTK3 + VTE instead of GTK4 + VTE.
(I'll fix vte3 later.)

> I think it's not appropriate for me to ask all these questions on an unrelated issue, but I don't want to spam your repo with minor issues and I appreciate your time. Thank you for your assistance with understanding

How about using https://github.com/rabbit-shocker/rabbit/discussions instead? I hope that we have non-outdated FAQ because we have many undocumented features.

**Comment by kingdonb** [comment]
My expression right now is 🤯 

It's working, I have been able to set up my fonts, this is great 🏆 

I will definitely make a fan post of all my favorite undocumented features and put it on Discussions, thanks so much!
